src/components/sidebar/sidebar.test.tsx

```diff
@@ -0,0 +1,321 @@
+import React from "react";
+import {
+  render,
+  screen,
+  waitFor,
+  waitForElementToBeRemoved,
+} from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import {
+  testStyledSystemPadding,
+  testStyledSystemWidth,
+} from "../../__spec_helper__/__internal__/test-utils";
+import CarbonProvider from "../carbon-provider";
+import Form from "../form";
+
+import Sidebar, { SidebarProps } from ".";
+
+const AllTheProviders = ({ children }: { children: React.ReactNode }) => (
+  <CarbonProvider validationRedesignOptIn>{children}</CarbonProvider>
+);
+
+beforeEach(() => {
+  jest.useFakeTimers();
+});
+
+afterEach(() => {
+  jest.runOnlyPendingTimers();
```

Comment:
**question**: is it necessary to run the pending timers after each test? I was curious about this, finding no use of `setTimeout` or `setInterval` in any related component other than in `FocusTrap`, where it's used in the `setElementFocus` method, where it wouldn't feel right to wait for this *after* a test rather than during it.

Because of this, I tried running all the `sidebar` tests with this line commented out, expecting at least some of them to fail - but they all passed. I'd rather not include this line unless we know it's necessary.

Reply:
I included the timer flushing because we use React Transition Group for handling CSS transitions, which relies on timers for state updates. 

Given that it’s a third-party package, I felt it was a best practice to flush any potential running timers. This precaution helps prevent any unexpected behavior from scheduled tasks leaking into other tests.

Reply:
That's a good point that I hadn't thought of. Happy to do this if it's a general best-practice.